protocols/imap/src/main/java/org/apache/james/imap/processor/base/SelectedMailboxImpl.java

```diff
@@ -362,37 +369,34 @@ public synchronized void event(Event event) {
     private void mailboxEvent(MailboxEvent mailboxEvent) {
         // Check if the event was for the mailbox we are observing
         if (mailboxEvent.getMailboxId().equals(getMailboxId())) {
-            MailboxSession.SessionId eventSessionId = mailboxEvent.getSessionId();
-            if (mailboxEvent instanceof MessageEvent) {
-                final MessageEvent messageEvent = (MessageEvent) mailboxEvent;
-                if (messageEvent instanceof Added) {
-                    final Collection<MessageUid> uids = ((Added) mailboxEvent).getUids();
-                    handleAddition(uids);
-                } else if (messageEvent instanceof FlagsUpdated) {
-                    FlagsUpdated updated = (FlagsUpdated) messageEvent;
-                    handleFlagsUpdates(mailboxEvent, eventSessionId, (FlagsUpdated) messageEvent, updated);
-                } else if (messageEvent instanceof Expunged) {
-                    handleMailboxExpunge(messageEvent);
-                }
-            } else if (mailboxEvent instanceof MailboxDeletion) {
-                handleMailboxDeletion(eventSessionId);
-            }
+            Match(mailboxEvent).of(
+                Case($(instanceOf(Added.class)),
+                    this::handleAddition),
+                Case($(instanceOf(FlagsUpdated.class)),
+                    this::handleFlagsUpdates),
+                Case($(instanceOf(Expunged.class)),
+                    this::handleMailboxExpunge),
+                Case($(instanceOf(MailboxDeletion.class)),
+                    this::handleMailboxDeletion)
+            );
         }
     }
 
-    private void handleMailboxDeletion(MailboxSession.SessionId eventSessionId) {
-        if (eventSessionId != sessionId) {
+    private Void handleMailboxDeletion(MailboxDeletion mailboxDeletion) {
+        if (mailboxDeletion.getSessionId() != sessionId) {
             isDeletedByOtherSession = true;
         }
+        return VOID;
```

Comment:
Can you explain why you return `null` here? (and others?)

Reply:
vavr pattern-matching requires a return value.
you can't have `void` methods.
So workaround is the have a `Void` method (with a capital letter) and return null as the only valid `Void` value.